File: teste.py
from sys import argv
script, filenameIn, filenameOut = argv
bases = 0
q30 = 0
q20 = 0
medio = 0
i = count = 0
with open(filenameIn) as f:
	for line in f:
		line = line.strip("\n")
		if(i==1):
			maior = len(line)
			menor = len(line)	
		if(count==1):
			if(len(line)>maior):
				maior = len(line)
			if(len(line)<menor):
				menor = len(line)
			bases = bases+len(line)
		if(count==3):
			for letter in line:
				if(ord(letter)>30):
					q30 = q30+1
				if(ord(letter)>20):
					q20 = q20+1
			count=-1
		i = i+1
		count = count+1
reads = i/4
medio = bases/reads
file = open(filenameOut, 'a+')
file.write("%f\t %f\t %d\t %d\t %f\t %d\t %d\n" % (reads, bases, q30, q20, medio, maior, menor))
file.close()

File: teste.go
package main

import (
	"os"
	"fmt"
	"bufio"
	"strconv"
)
var text string
var i,bases,count,max,min,medio, q20, q30 int
func main(){
	args := os.Args[1:]
	fileIn, err := os.Open(args[0])
	if err != nil{
		fmt.Println(err)
		return	
	}
	sc := bufio.NewScanner(fileIn)
	for sc.Scan(){
		text = sc.Text()	
		if count==1 {
			if i==1 {
				max = len(text)
				min = len(text)
			}else{
				if len(text) > max {
					max = len(text)
				}
				if len(text) < min {
					min = len(text)
				}
			}
			bases = bases+len(text)
		}
		if count == 3 {
			for _, r := range text{
				if r > 20{
					q20 = q20+1
				}
				if r > 30{
					q30 = q30+1
				}
			}
			count= -1
		}
		count++
		i++
	}
	total := i/4
	medio = bases/total
	fileIn.Close()
	fileOut, err := os.Create(args[1])	
	result := strconv.Itoa(total)+"\t"+strconv.Itoa(bases)+"\t"+strconv.Itoa(q30)+"\t"+strconv.Itoa(q20)+"\t"+strconv.Itoa(medio)+"\t"+strconv.Itoa(max)+"\t"+strconv.Itoa(min)
	n,_ := fileOut.WriteString(result)
	fmt.Println("wrote ",n)
}

#####
